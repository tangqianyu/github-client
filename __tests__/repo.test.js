const { default: Repo } = require("../models/repo");
import RepoComponent from '../components/repo'
import renderer from 'react-test-renderer';
import React from 'react';

it("model test", () => {
  const repo = new Repo({
    name: "hello",
    description: "world",
    owner: { login: "test" },
  });
  expect(repo.name).toBe("hello");
  expect(repo.description).toBe("world");
  expect(repo.username).toBe("test");

});

test('renders correctly', () => {
  const repo = {
    name: "hello",
    description: "world",
    owner: { login: "test" },
  };
  const tree = renderer.create(<RepoComponent repo={repo} />).toJSON();
  expect(tree).toMatchSnapshot();
});



