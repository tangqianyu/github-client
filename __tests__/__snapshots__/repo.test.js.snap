// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#333",
      "borderWidth": 1,
      "height": 50,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View>
      <Text>
        Repo Name: 
      </Text>
    </View>
    <View>
      <Text>
        hello
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Text>
      Owner Name: 
    </Text>
    <Text>
      test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Text>
      Description: 
    </Text>
    <Text>
      world
    </Text>
  </View>
</View>
`;
